### PR TITLE
Add tic-tac-toe example to workspace

### PR DESCRIPTION
**Pull request by hayes-mysten**
## Description 

Describe the changes or additions included in this PR.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: sdk/typescript/test/e2e/object-vector.test.ts [hunk 1]
```diff
@@ -62,16 +62,9 @@ describe('Test Move call with a vector of objects as input', () => {
...
-		async () => {
-			await destroyObjects([await mintObject(7), await mintObject(42)], /* withType */ true);
-		},
-		{
-			// TODO: This test is currently flaky, so adding a retry to unblock merging
-			retry: 10,
-		},
-	);
+	it('Test object vector with type hint', async () => {
+		await destroyObjects([await mintObject(7), await mintObject(42)], /* withType */ true);
```
Comment (hayes-mysten) [inline sdk/typescript/test/e2e/object-vector.test.ts:66]: I think the I fixed the thing that caused this to be flakey before...

File: examples/tic-tac-toe/ui/src/App.tsx
Comment (amnn) [inline examples/tic-tac-toe/ui/src/App.tsx:1]: Something funky going on here -- it looks like the formatting is now *not* following the prettier config at the root of the `sui` repo. (@manolisliolios mentioned this was an issue in his editor as well -- on my end, I was manually calling `prettier --config ~/sui/prettier.config.js -w .`).

**Review by amnn — COMMENTED** [review]
Thanks for doing this @hayes-mysten ! I'm happy either way, I just couldn't get it to work in the workspace because `vite` seemed unable to find `@mysten/bcs` and `@mysten/wallet-standard` in this set-up -- were you able to try that out and confirm that it works?

Also, it looks like something funky is happening with the formatting -- I'm pretty sure it was correct before (conforming to our prettier config), but now it's not -- the additions in this PR introduce spaces instead of tabs, for instance.

File: examples/tic-tac-toe/ui/src/App.tsx
Comment (hayes-mysten) [reply examples/tic-tac-toe/ui/src/App.tsx:1]: There was a custom prettier.config in the ui directory, which I just removed.  Good catch

**Comment by hayes-mysten** [comment]
> vite seemed unable to find @mysten/bcs and @mysten/wallet-standard in this set-up -- were you able to try that out and confirm that it works?

Yes, this seems to work fine for me when I run it locally

File: examples/tic-tac-toe/ui/src/env.testnet.ts
Comment (amnn) [inline examples/tic-tac-toe/ui/src/env.testnet.ts:1]: These are generated by `scripts/publish.sh` -- could you update the template there too? (Out of interest, what is the difference between the two approaches -- asking as a TS noob).

**Review by amnn — APPROVED** [review]
Thanks for cleaning this all up as well! Accepting to unblock, but also want to get @Jordan-Mysten's thoughts on whether we should fold this into the workspace.

File: examples/tic-tac-toe/ui/src/env.testnet.ts
Comment (hayes-mysten) [reply examples/tic-tac-toe/ui/src/env.testnet.ts:1]: they are the same, we just have a lint rule enforcing that doesn't like the anonymous export.  I think there might be some slight differences in error messages or something that can be produced, I can't recall off the top of my head why this is preferred, I was just making the linter happy

**Pull request merged**